src/pages/instances/actions/OpenConsoleBtn.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { FC } from "react";
+import { useNavigate } from "react-router-dom";
+import { LxdInstance } from "types/instance";
+import { Button } from "@canonical/react-components";
+
+interface Props {
+  instance: LxdInstance;
+}
+
+const OpenConsoleBtn: FC<Props> = ({ instance }) => {
+  const navigate = useNavigate();
+
+  const handleOpen = () => {
+    if (instance.type === "virtual-machine" && instance.status === "Running") {
+      navigate(
+        `/ui/${instance.project}/instances/detail/${instance.name}/vga-console`
+      );
+    } else {
+      navigate(
+        `/ui/${instance.project}/instances/detail/${instance.name}/text-console`
+      );
+    }
+  };
+
+  return (
+    <Button
+      appearance="base"
+      dense
+      hasIcon
+      onClick={handleOpen}
+      title="Console"
+    >
+      <i className="p-icon--canvas">Open VGA Terminal</i>
```

Comment:
We should probably use react-component's `Icon`, here but also in all the other 22 places where we still use the `i` tag. It's out of the scope of this PR: I'll file a JIRA ticket about this.

Reply:
I can fix it here and keep it in mind for the future.

Reply:
Sorry, again on this point - should the "Open VGA Terminal" be set as `aria-label` of the `Button` here? And have the `<Icon />` self-closing then?

Reply:
Yes, this is the way.